File: KTC 2023 Dataset/Codes_Python/Codes_Python/Main_TrainingData.py
import scipy as sp
import main
import KTCScoring
import matplotlib.pyplot as plt
# import warnings


main.main('TrainingData', 'Output', 3)

score = 0
for ii in range(4):

    reco = sp.io.loadmat('Output/' + str(ii+1) + '.mat')
    reco = reco["reconstruction"]
    truth = sp.io.loadmat('GroundTruths/true' + str(ii+1) + '.mat')
    truth = truth["truth"]

    s = KTCScoring.scoringFunction(truth, reco)
    print('Score from target ' + str(ii+1) + ' = ' + str(s))
    score = score + s

print('Final score: ' + str(score) + ' / 4.00')

File: KTC 2023 Dataset/Codes_Matlab/Codes_Matlab/Main_TrainingData.m
%This script runs the main.m function for input file 'TrainingData', and
%then assesses all the output reconstructions using the given scoring
%algorithm.
close all; clear all; clc;

path(path,'MiscCodes/')
main('TrainingData', 'Output', 1);

score = 0;
for ii = 1:4

    load(['Output/' num2str(ii) '.mat']);
    load(['GroundTruths/true' num2str(ii) '.mat']);

    s = scoringFunction(truth, reconstruction);
    disp(['Score from target ' num2str(ii) ' = ' num2str(s)])
    score = score + s;

end

disp(['Final score: ' num2str(score/4) ])

